utils/temptablecache.go

```diff
@@ -0,0 +1,99 @@
+package utils
+
+import (
+	"sync"
+)
+
+// TempTableCache is a thread-safe cache for storing temporary table mappings
+var TempTableCache = &CacheSimple{
+	data:        make(map[string]interface{}),
+	maxSize:     10,
+	accessOrder: make([]string, 0),
+}
+
+// Defines a thread-safe in-memory cache
+type CacheSimple struct {
+	data        map[string]interface{}
+	mu          sync.RWMutex
+	maxSize     int
+	accessOrder []string // Keeps track of insertion order for cleanup
+}
+
+// Retrieves a value from the cache
+func (c *CacheSimple) Get(key string) (interface{}, bool) {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	value, exists := c.data[key]
+	// If the key already exists, update its value and move it to the end of the access order
+	if exists {
+		c.data[key] = value
```

Comment:
Why is this line necessary if we just got the value from the cache?

Reply:
good point. Removed